assets/js/blocks/cart-checkout/cart/block.js

```diff
@@ -1,22 +1,51 @@
 /**
  * External dependencies
  */
+import { CART_STORE_KEY as storeKey } from '@woocommerce/block-data';
+import { dispatch } from '@wordpress/data';
 import { useStoreCart } from '@woocommerce/base-hooks';
-import { RawHTML } from '@wordpress/element';
+import { useEffect, RawHTML } from '@wordpress/element';
 import LoadingMask from '@woocommerce/base-components/loading-mask';
 import {
 	ValidationContextProvider,
 	CartProvider,
 } from '@woocommerce/base-context';
+import { translateJQueryEventToNative } from '@woocommerce/base-utils';
+import withScrollToTop from '@woocommerce/base-hocs/with-scroll-to-top';
 
 /**
  * Internal dependencies
  */
 import FullCart from './full-cart';
 
-const Block = ( { emptyCart, attributes } ) => {
+// Make it so we can read jQuery events triggered by WC Core elements.
+translateJQueryEventToNative( 'added_to_cart', 'wc-blocks_added_to_cart' );
+
+const Block = ( { emptyCart, attributes, scrollToTop } ) => {
 	const { cartItems, cartIsLoading } = useStoreCart();
 
+	useEffect( () => {
+		const invalidateCartData = () => {
+			if ( cartItems.length === 0 ) {
+				dispatch( storeKey ).invalidateResolutionForStore();
+				scrollToTop();
+			}
+		};
```

Comment:
Nifty! While this works great for cartItems removed from the block, I wonder if we should also add an event listener for when an item is removed from the cart js _widget_ and show the empty cart placeholder when the items in the cart are empty? If you agree, definitely can be done in another issue (I just thought it would be a good use of the jQuery translation layer as well)

Reply:
Good point, done in 4cfb838be992f6a4b25bcaf8d80129e408097c95.

The Cart widget was not working for me in the Cart page, so to test I needed to create a post with the Cart block and the Cart widget in the sidebar:

![Peek 2020-07-10 12-07](https://user-images.githubusercontent.com/3616980/87143208-fe4be300-c2a5-11ea-9edb-bf1e56ec26f1.gif)